services/core-web/src/components/common/CoreTable.tsx

```diff
@@ -94,29 +94,26 @@ const CoreTable = <T,>(props: CoreTableProps<T>) => {
     );
   };
 
-  const expansionProps = expandProps
-    ? {
-        rowExpandable:
-          expandProps.rowExpandable ?? ((record) => expandProps.getDataSource(record).length > 0),
-        expandIcon: renderTableExpandIcon,
-        expandRowByClick: true,
-        expandedRowRender: expandProps.expandedRowRender ?? renderExpandedRow,
-        ...expandProps,
-      }
-    : null;
-
-  const matchChildColumnsToParentProps = expandProps
-    ? {
-        expandIcon: renderTableExpandIcon,
-        indentSize: 0,
-      }
-    : null;
+  const getExpansionProps = () => {
+    if (expandProps) {
+      return expandProps.matchChildColumnsToParent
+        ? { expandIcon: renderTableExpandIcon, indentSize: 0, ...expandProps }
+        : {
+            rowExpandable:
+              expandProps.rowExpandable ??
+              ((record) => expandProps.getDataSource(record).length > 0),
+            expandIcon: renderTableExpandIcon,
+            expandRowByClick: true,
+            expandedRowRender: expandProps.expandedRowRender ?? renderExpandedRow,
+            ...expandProps,
+          };
+    }
+    return { showExpandColumn: false };
```

Comment:
ran into an issue where if the dataSource had children, it would show the expand icon even if I didn't want it to. Forced it to be deliberate.